src/join.t.sol

```diff
@@ -0,0 +1,1237 @@
+                );
+            }
+        }
+
+        // We have failed if we reach here
+        assertTrue(false);
+    }
+
+    function giveTokens(address token, uint256 amount) internal {
+        // Edge case - balance is already set for some reason
+        if (GemLike(token).balanceOf(address(this)) == amount) return;
+
+        // Scan the storage for the balance storage slot
+        for (uint256 i = 0; i < 200; i++) {
+            // Solidity-style storage layout for maps
+            {
+                bytes32 prevValue = hevm.load(
+                    address(token),
+                    keccak256(abi.encode(address(this), uint256(i)))
+                );
+
+                hevm.store(
+                    address(token),
+                    keccak256(abi.encode(address(this), uint256(i))),
+                    bytes32(amount)
+                );
+                if (GemLike(token).balanceOf(address(this)) == amount) {
+                    // Found it
+                    return;
+                } else {
+                    // Keep going after restoring the original value
+                    hevm.store(
+                        address(token),
+                        keccak256(abi.encode(address(this), uint256(i))),
+                        prevValue
+                    );
+                }
+            }
+
+            // Vyper-style storage layout for maps
+            {
+                bytes32 prevValue = hevm.load(
+                    address(token),
+                    keccak256(abi.encode(uint256(i), address(this)))
+                );
+
+                hevm.store(
+                    address(token),
+                    keccak256(abi.encode(uint256(i), address(this))),
+                    bytes32(amount)
+                );
+                if (GemLike(token).balanceOf(address(this)) == amount) {
+                    // Found it
+                    return;
+                } else {
+                    // Keep going after restoring the original value
+                    hevm.store(
+                        address(token),
+                        keccak256(abi.encode(uint256(i), address(this))),
+                        prevValue
+                    );
+                }
+            }
+        }
+
+        // We have failed if we reach here
+        assertTrue(false, "TestError/GiveTokens-slot-not-found");
+    }
+
+    function testGemJoin_REP() public {
+        giveTokens(address(rep), 100 * WAD);
+        GemJoin repJoin = new GemJoin(address(vat), "REP", address(rep));
+        assertEq(repJoin.dec(), 18);
+        giveAuth(address(vat), address(repJoin));
+
+        rep.approve(address(repJoin), uint256(-1));
+        assertEq(rep.balanceOf(address(this)), 100 ether);
+        assertEq(rep.balanceOf(address(repJoin)), 0);
+        assertEq(vat.gem("REP", address(this)), 0);
+        repJoin.join(address(this), 10 ether);
+        assertEq(rep.balanceOf(address(repJoin)), 10 ether);
```

Comment:
Missing:
`assertEq(rep.balanceOf(address(this)), 90 ether);`

Reply:
Seems to happen everywhere.

Reply:
I just did a [diffcheck](https://www.diffchecker.com/JAW3osv3) against `dss-gem-joins` `join.t.sol` and it seems the check was also missing there, as discussed in chat.

Reply:
Done in 883d8d2